### DISCUSSION TITLE
WebUI Types

### DISCUSSION THREAD

**Discussion by huntabyte** [open-webui/open-webui#2639] [General]
**Is your feature request related to a problem? Please describe.**
When running the project during local development, the types in the frontend portion of the application are often unclear.

**Describe the solution you'd like**
Type safety throughout the application that enables changes to be made with confidence.

**Describe alternatives you've considered**
N/A

**Additional context**
I can assist with some PRs :)

**Comment by tjbck** [comment]
Feel free to make a PR! Just a warning this project is still being actively developed and a lot of the code will be either replaced in favour of other implementations or removed entirely, so it's probably best if you focus on only the core part of the code. Good luck!

**Reply by huntabyte** [reply]
Gotcha! Do you have a rough idea as to what you'd consider a core part vs non-core part of the code?
